src/network/networkpacket.cpp

```diff
@@ -72,8 +72,8 @@ char* NetworkPacket::getString(u32 from_offset)
 
 void NetworkPacket::putRawString(const char* src, u32 len)
 {
-	if (m_read_offset + len * sizeof(char) >= m_datasize) {
-		m_datasize += len * sizeof(char);
+	if (m_read_offset + len * sizeof(char) > m_datasize) {
+		m_datasize = m_read_offset + len * sizeof(char);
```

Comment:
According to the ISO/IEC C++ standard section 5.3.3.1 sizeof(char) always evaluates to 1.

I left those in there because I figured someone wanted them there for stylistic reasons.